content/en/blog/_posts/2019-08-28-cloud-native-java-controller-sdk.md

```diff
@@ -0,0 +1,102 @@
+
+---
+layout: blog
+title: "Cloud Native Java: Develop kubernetes controller application"
+date: 2019-08-28
+slug: Cloud-Native-Java-Develop-Kubernetes-Controller-Application
+---
+
+**Authors:** Min Kim (Ant Financial), Tony Ado (Ant Financial)
+
+The [Kubernetes offical Java SDK](https://github.com/kubernetes-client/java) project 
+recently released their latest work on providing the Java Kubernetes developers 
+a handy Kubernetes controller-builder SDK which is helpful for easily developing 
+advanced workloads or systems.  
+
+## Overall
+
+Java is no doubt one of the most popular programming languages in the world but
+it's been difficult for a period time for those non-Golang developers to build up 
+their customized controller/operator due to the lack of library resources in the 
+community. In the world of Golang, there're already some excellent controller 
+frameworks, for example, [controller runtime](https://github.com/kubernetes-sigs/controller-runtime),
+[operator SDK](https://github.com/operator-framework/operator-sdk) and all these
+existing Golang frameworks are relying on the various utilities from the 
+[Kubernetes Golang SDK](https://github.com/kubernetes/client-go) proven to 
+be stable over years. Driven by the emerging need of further integration into 
+the platform of Kubernetes, we not only ported many essential toolings from Golang 
+SDK into the kubernetes Java SDK including informers, work-queues, leader-elections, 
+etc. but also developed a controller-builder SDK which wires up everything into 
+a runnable controller without hiccups.
+
+
+## Backgrounds
+
+Before we dig further, we have to know why we probably need Java in Kubernetes:
+
+- __Integrating legacy enterprise Java systems__: Many companies have their legacy 
+systems or frameworks written in Java in favor of stability. We are not able to 
+move everything to Golang straightly.
```

Comment:
straightly? I'm not sure what you mean here.

Reply:
i replaced w/ "easily"